src/server/document-controller.js

```diff
@@ -74,19 +74,24 @@ class DocumentController {
       if (!templateDocument) {
         throw new NotFound();
       }
+
+      const templateDocumentRoom = templateDocument.roomId ? await this.roomService.getRoomById(templateDocument.roomId) : null;
+      if (templateDocumentRoom && !isRoomOwnerOrMember({ room: templateDocumentRoom, userId: user._id })) {
```

Comment:
We should check for `isRoomOwnerOrCollaborator` here rather than `isRoomOwnerOrMember`, right?

Reply:
I was thinking long about that, but then decided not to, because then we would have to also change line 88 further down to:

~~~
if (room && view === 'edit' && !isRoomOwnerOrCollaborator({ room, userId: user._id })) {
  throw new Forbidden();
}

if (room && view !== 'edit' && !isRoomOwnerOrMember({ room, userId: user._id })) {
  throw new Forbidden();
}
~~~

So I kept the paradigm: "If the user is allowed to see it (as oppposed to: edit it), they will see it. If they try to save changes, they will get an error message"

Reply:
I saw the discrepancy there but I though one case is a guard for viewing ("read") and another for cloning ("write") 🤔 However I agree that the members still have a right so view even with cloning, just not to write, which is not the scope of this controller endpoint. In the end we're talking about a very edge case anyway, so I agree then, let's keep it so.